LONDON | Pezhman Azizi | Module-User-Focused-Data | Wireframe | Week 1 | Sprint 1

Pezhman-Azizi opened the pull request:
<!--

You must title your PR like this:

COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK

For example,

NW4 | Carol Owen | HTML-CSS-Module | Week1

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://syllabus.codeyourfuture.io/guides/code-style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

1. Added a structured layout for an introductory page on "Git and GitHub Fundamentals."
2. Implemented a responsive footer with social links, a newsletter subscription form, and quick links.
3. Styled the header, main content, and footer to enhance readability and visual appeal.

## Questions

Mobile Layout :
Does the mobile layout work well for readability, especially in the articles section? If not, would you suggest a quick change, like switching to a single-column layout for a better mobile experience?


Ask any questions you have for your reviewer.

Pezhman-Azizi committed:
fixing closing tag

Pezhman-Azizi committed:
fix: update CSS file path for proper styling

Pezhman-Azizi committed:
changed the title of the page

Pezhman-Azizi committed:
add main heading and introduction text to HTML

Pezhman-Azizi committed:
add header section with Git and GitHub logo image

Pezhman-Azizi committed:
add main content section with articles on Fork, Clone, and Branch

Pezhman-Azizi committed:
add footer with social media links for GitHub, Instagram, Facebook, and Twitter

Pezhman-Azizi committed:
added newsletter sign-up form to footer section

Pezhman-Azizi committed:
added quick links section to footer for easy navigation

Pezhman-Azizi committed:
added copyright notice to footer

Pezhman-Azizi committed:
added general styles for body and html elements

Pezhman-Azizi committed:
added styling for title and description elements

Pezhman-Azizi committed:
added styling for header image to adjust width and height

Pezhman-Azizi committed:
added article styles for layout, images, and links in main section

Pezhman-Azizi committed:
added footer styles for layout, social links, sign-up form, and links

SallyMcGrath reviewed (changes requested):
Thanks for this @Pezhman-Azizi . It's a good start 🎉 

There are a couple of problems that I can help with. 

1) the fixed footer is covering over the content so I can't see it
2) when I uncovered the content, it was three boxes in a line, which doesn't quite match the wireframe

Do you need help with this? Come to class with your work to get help.

To answer your q, yes, you would typically switch to a single column layout for mobile. There are a few ways to approach this. Older frameworks use media queries to set different styles for mobile. More modern CSS uses auto flow grids. Take a look at https://web.dev/articles/one-line-layouts

Pezhman-Azizi committed:
fixed the footer covering the article

Pezhman-Azizi committed:
applied the first-child rule

Pezhman-Azizi committed:
applied the :first-child rule to the article

Pezhman-Azizi committed:
commented out the third article so it matches the original wire frame

Pezhman-Azizi committed:
web site rights are placed correctly in the footer

Pezhman-Azizi commented:
Thank you so much for your feedback! I made the adjustments to the footer positioning so it doesn’t cover content, and I updated the layout to switch to a single-column view on smaller screens. I also implemented media queries and modern CSS techniques, as you suggested, to improve the responsiveness of the design. Your guidance was clear and easy to follow, and it really helped make my project more user-friendly.

Thanks again for the support!

SallyMcGrath commented on Wireframe/index.html:
```diff
@@ -42,36 +44,36 @@ <h2>Branch</h2>
-            <a href="https://facebook.com/" target="_blank" aria-label="Facebook"><i class="fab fa-facebook-f"></i></a>
-            <a href="https://twitter.com/" target="_blank" aria-label="Twitter"><i class="fab fa-twitter"></i></a>
-          </div>
+            <!-- Social Links Column -->
```
Instead of a div, what else could you use to signal a list of navigation links?

Pezhman-Azizi replied:
Hi Sally,
Oh they could actually be replaced with <ul> and <li> elements. You are right. The HTML file has got more semantic style.
In order to do so, I’ve updated the code by replacing the `div` with a `<ul>` for the social links, as suggested. The links are now properly structured and styled. Thanks for pointing that out!

Let me know if it looks good.

SallyMcGrath commented on Wireframe/css/style.css:
```diff
@@ -191,20 +191,6 @@ footer p {
   font-size: 0.8em;
```
Don't make font tiny. Most people cannot read small fonts. We make websites for other people to use.

Pezhman-Azizi replied:
I’ve adjusted the font sizes to make sure they’re more readable and accessible for all users. Thanks for the reminder to keep usability in mind!

Please let me know if you think the new sizes work better.

SallyMcGrath reviewed (commented):
Nice changes.

ChatGPT writes comments in that style. Have you noticed?

Pezhman-Azizi committed:
Implemented semantic structure for social links and update footer styling
